File: day1.py
#this script solves the problem here ----> https://adventofcode.com/2021/day/1
import sys
#reads the 1st command line argument, which should be the input file, and 
inp = sys.argv[1]
with open(inp,'r') as file:
    measurements = file.readlines()

measurements = [int(m) for m in measurements if m]

#loops through and counts how many times measurements increase from one reading to the next
increases = 0
for i in range(1,len(measurements)):
    if measurements[i] > measurements[i-1]: increases += 1

print(f'The measurement increased {increases} times')

File: day1.c
//this script solves the problem here ----> https://adventofcode.com/2021/day/1
#include<stdio.h>
int measurements[] = {199,200,208,210,200,207,240,269,260,263};

int main(){
    int loops = (sizeof measurements)/4;

    //loops through list to see how many times the measurement increases
    int count = 0;
    for (int i = 1; i < loops; i++)
    {
        if (measurements[i] > measurements[i-1]){
            count++;
        }
    }

    //loops through the list again to see how many times the group of three measurements increases
    int count2 = 0;
    loops = loops;
    for (int i = 3; i < loops; i++)
    {
        if (measurements[i] + measurements[i-1] + measurements[i-2] > measurements[i-1] + measurements[i-2] + measurements[i-3]){
            count2++;
        }
    }

    //prints answers for parts 1 and 2 of the problem
    printf("ans: %d\n", count);
    printf("ans2: %d\n",count2);
    return 0;
}